node-js-basics-101/node_modules/uws/uws.js

```diff
@@ -0,0 +1,563 @@
+'use strict';
+
+const http = require('http');
+const EventEmitter = require('events');
+const EE_ERROR = 'Registering more than one listener to a WebSocket is not supported.';
+const DEFAULT_PAYLOAD_LIMIT = 16777216;
+
+function noop() {}
+
+function abortConnection(socket, code, name) {
+    socket.end('HTTP/1.1 ' + code + ' ' + name + '\r\n\r\n');
+}
+
+function emitConnection(ws) {
+    this.emit('connection', ws);
+}
+
+function onServerMessage(message, webSocket) {
+    webSocket.internalOnMessage(message);
+}
+
+const native = (() => {
+    try {
+        try {
+            return process.binding('uws_builtin');
+        } catch (e) {
+            return require(`./uws_${process.platform}_${process.versions.modules}`);
+        }
+    } catch (e) {
+        const version = process.version.substring(1).split('.').map(function(n) {
+            return parseInt(n);
+        });
+        const lessThanSixFour = version[0] < 6 || (version[0] === 6 && version[1] < 4);
+
+        if (process.platform === 'win32' && lessThanSixFour) {
+            throw new Error('µWebSockets requires Node.js 6.4.0 or greater on Windows.');
+        } else {
+            throw new Error('Compilation of µWebSockets has failed and there is no pre-compiled binary ' +
+            'available for your system. Please install a supported C++11 compiler and reinstall the module \'uws\'.');
+        }
+    }
+})();
+
+native.setNoop(noop);
+
+var _upgradeReq = null;
+
+const clientGroup = native.client.group.create(0, DEFAULT_PAYLOAD_LIMIT);
+
+native.client.group.onConnection(clientGroup, (external) => {
+    const webSocket = native.getUserData(external);
+    webSocket.external = external;
+    webSocket.internalOnOpen();
+});
+
+native.client.group.onMessage(clientGroup, (message, webSocket) => {
+    webSocket.internalOnMessage(message);
+});
+
+native.client.group.onDisconnection(clientGroup, (external, code, message, webSocket) => {
+    webSocket.external = null;
+
+    process.nextTick(() => {
+        webSocket.internalOnClose(code, message);
+    });
+
+    native.clearUserData(external);
+});
+
+native.client.group.onPing(clientGroup, (message, webSocket) => {
+    webSocket.onping(message);
+});
+
+native.client.group.onPong(clientGroup, (message, webSocket) => {
+    webSocket.onpong(message);
+});
+
+native.client.group.onError(clientGroup, (webSocket) => {
```

Comment:
__[jshint]__
* 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

<!-- Comment by [SideCI](https://sideci.com) -->